proxy/cache.go

```diff
@@ -161,6 +164,18 @@ func getTTLIfLower(h *dns.RR_Header, ttl uint32) uint32 {
 	return ttl
 }
 
+// Updates a given TTL to fall within the range specified
+// by the cacheMinTTL and cacheMaxTTL settings
+func respectTTLOverrides(ttl uint32, cacheMinTTL uint32, cacheMaxTTL uint32) uint32 {
+	if cacheMinTTL > 0 && ttl < cacheMinTTL {
+		return cacheMinTTL
+	}
+	if cacheMaxTTL > 0 && ttl > cacheMaxTTL {
```

Comment:
let's use `cacheMaxTTL != 0` as cacheMaxTTL can't be <0

Reply:
Done.